solver.py

```diff
@@ -122,7 +122,7 @@ def _get_msg_entities():
             # Store a index-constraint mapping
             from collections.abc import Iterable
 
-            indices = c.create_slvs_data(self.solvesys, group=group)
+            indices = c.py_data(self.solvesys, group=group)
```

Comment:
That's quite the fundamental change here, is it really needed for this PR? I would rather have this in a separate PR and apply it also to entities.

But i'm not actually sure this makes sense, The slvs solve system cannot be created and updated at a later point. So a new one is created everytime the solver runs. If we don't explicitly recreate slvs data py_data might return handles from the last solver run.

Reply:
I am going to think of a different way to do it so it doesn't have that many changes to the higher level abstractions

Reply:
@hlorus 
Either:
- I rename the `create_slvs_data()` in all the constraints and entities that use it.
- I write the "empty return" return step in all the `create_slvs_data()` for the dimensional constraints.

The reason I refactored `create_slvs_data()` to `py_data()` in the higher level abstraction is because the `.py_data` attribute is used by the entities to pass on the solver handles.

I can create a separate PR now that refactors all the entities and constraints to 
- have a `py_data()` function that serves as interface and returns the slvs handles.
  > That function should only be implemented by the abstract/mother classes like `GenericX` 
- have a private function `create_slvs_data()` that generates the data.
  > This function is implemented by all the daughter classes to create handles specific to themselves

I would also suggest a name change for those functions, for example `solver_handles()` and `generate_solver_handles()`